client/src/modules/shopping-list/model/reducer.js

```diff
@@ -22,10 +22,10 @@ const items = (state, action) => {
             : product
         )
       };
-    case ItemActionTypes.VOTE_FOR_ITEM_SUCCESS:
+    case ItemActionTypes.VOTE_FOR_SUCCESS:
       return {
         ...state,
-        products: [...state.products].map(product =>
+        products: state.products.map(product =>
```

Comment:
why did you removed spread from here? I remember that there was some problem that react didn't saw changes in state and was not re-rendering the app. Did you test it??

Reply:
Map returns new array, I tested it, everything works good